62_CAD/shaders/main_pipeline/vertex_shader.hlsl

```diff
@@ -92,6 +94,51 @@ PSInput main(uint vertexID : SV_VertexID)
     // ~~Later, most likely We will require pulling all 3 vertices of the triangle, that's where you need to know which triangle you're currently on, and instead of objectID = vertexID/4 which we currently do, you will do vertexID/3 and pull all 3 of it's vertices.~~
     // Ok, brainfart, a vertex can belong to multiple triangles, I was thinking of AA but triangles share vertices, nevermind my comment above.
 
+#define DTM
+#ifdef DTM
+    PSInput outV;
+
+    // Default Initialize PS Input
+    outV.position.zw = float2(0.0, 1.0);
+    outV.data1 = uint4(0, 0, 0, 0);
+    outV.data2 = float4(0, 0, 0, 0);
+    outV.data3 = float4(0, 0, 0, 0);
+    outV.data4 = float4(0, 0, 0, 0);
+    outV.interp_data5 = float2(0, 0);
+    outV.setObjType(ObjectType::TRIANGLE_MESH);
+    outV.setMainObjectIdx(pc.triangleMeshMainObjectIndex);
+
+    TriangleMeshVertex vtx = vk::RawBufferLoad<TriangleMeshVertex>(pc.triangleMeshVerticesBaseAddress + sizeof(TriangleMeshVertex) * vertexID, 4u);
+    pfloat64_t2 vtxPos;
+    vtxPos.x = _static_cast<pfloat64_t>(vtx.pos.x);
+    vtxPos.y = _static_cast<pfloat64_t>(vtx.pos.y);
+
+    MainObject mainObj = mainObjects[pc.triangleMeshMainObjectIndex];
+    ClipProjectionData clipProjectionData = getClipProjectionData(mainObj);
+
+    float2 transformedPos = transformPointScreenSpace(clipProjectionData.projectionToNDC, globals.resolution, vtxPos);
+
+    outV.position.xy = transformedPos;
+    outV.position = transformFromSreenSpaceToNdc(outV.position.xy, globals.resolution);
+    outV.setHeightAtMeshVertex(vtx.height);
```

Comment:
I need to think about this a little more. my intuition says height being in `float` is just fine.
So you could just cast the height value to float here, no need to worry about dealing with `.z` being double.
in the gpu-driven thing I'm doing I might make this height atrribute of each vertex a contigous separate block in floats instead of doubles.

for now just use `pfloat64_t3` for your vertices and cast the height to float.